### PR TITLE
[openapi-gen] Add auth service to OpenAPI gen

### PR DESCRIPTION
**Pull request by jaylees14**
Adds the auth service to OpenAPI generator. 

I feel like the `Auth` trait should be moved somewhere else? Thoughts?

### CODE REVIEW

File: src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala [hunk 1]
```diff
@@ -22,6 +22,7 @@ private class OpenAPIGenerator private (name: String, version: String, descripti
   private val errorTracker = FlagMapView(
     400 -> generateError("Invalid request", "Invalid request parameters: name"),
     401 -> generateError("Valid request but forbidden by server", "Not authorised to create this object"),
+    403 -> generateError("Valid request but server will not fulfill", "User with this ID already exists"),
```
Comment (willjvsmith) [inline src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala:25]: This made me think, what if the user adds endpoints in their service, then updates the openAPI doc. It gets overwritten when regenerating 😱 do we need hooks for openAPI?!!!

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀 

Forgive me for being dense, but could you not continue use the `AuthMethod` from `Metadata`. I'm not familiar with the architecture being used to generate these yaml files.

**Comment by jaylees14** [comment]
> LGTM 🚀 
> 
> 
> 
> Forgive me for being dense, but could you not continue use the `AuthMethod` from `Metadata`. I'm not familiar with the architecture being used to generate these yaml files.

We could do, but we usually keep the generators separate from the AST so there is some level of decoupling. They usually have their own AST which gets mapped into in `ProjectBuilder`

File: src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala (hunk 1 above)
Comment (jaylees14) [reply src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala:25]: Good point.. no idea how we’d do that 😂

**Comment by willjvsmith** [comment]
> We could do, but we usually keep the generators separate from the AST so there is some level of decoupling. They usually have their own AST which gets mapped into in ProjectBuilder

I see, I reused AST types in the Go generation if they were exactly the same

**Review by Lewiky — APPROVED** [review]
Nicely done

**Pull request merged**